src/handler.rs

```diff
@@ -535,20 +548,193 @@ where
 
     fn get_inflight_fd(
         &mut self,
-        _inflight: &vhost::vhost_user::message::VhostUserInflight,
+        inflight: &vhost::vhost_user::message::VhostUserInflight,
     ) -> VhostUserResult<(vhost::vhost_user::message::VhostUserInflight, File)> {
-        // Assume the backend hasn't negotiated the inflight feature; it
-        // wouldn't be correct for the backend to do so, as we don't (yet)
-        // provide a way for it to handle such requests.
-        Err(VhostUserError::InvalidOperation)
+        let ret_val = -1;
+        // Total size of the inflight queue region
+        let mut total_mmap_size =
+            self.get_inflight_queue_size(inflight.queue_size) * inflight.num_queues as usize;
+
+        // Create a memfd region to hold the queues for inflight I/O tracking
+        let mut inflight_fd = -1;
+        let mmap_ptr = self.memfd_alloc(total_mmap_size, &mut inflight_fd)?;
+
+        self.inflight_file = Some(unsafe { File::from_raw_fd(inflight_fd as i32) });
+
+        if mmap_ptr == ret_val as *mut c_void {
+            total_mmap_size = 0;
+            self.inflight_mapping_addr = None;
+        } else {
+            unsafe { libc::memset(mmap_ptr, 0, total_mmap_size) };
+            self.inflight_mapping_addr = Some(GuestAddress::new(mmap_ptr as u64));
+        }
+        self.inflight_mmap_size = total_mmap_size;
+
+        Ok((
+            VhostUserInflight {
+                mmap_size: total_mmap_size as u64,
+                mmap_offset: 0,
+                num_queues: inflight.num_queues,
+                queue_size: inflight.queue_size,
+            },
+            unsafe { File::from_raw_fd(inflight_fd as RawFd) },
```

Comment:
We have already build an File object from `inflight_fd`, so it will cause double close of the underlying `inflight_fd` and then bad thing may happen. You should `dup(inflight_fd)` here.

Reply:
In other word, OOP based design is better here:)